src/Cache/CacheContent.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\View\Cache;
+
+use DateInterval;
+use InvalidArgumentException;
+use Yiisoft\Cache\CacheInterface;
+use Yiisoft\Cache\CacheKeyNormalizer;
+use Yiisoft\Cache\Dependency\Dependency;
+
+use function array_merge;
+use function get_class;
+use function gettype;
+use function is_string;
+use function is_object;
+use function sprintf;
+use function strtr;
+
+/**
+ * CacheContent caches content, supports the use of dynamic content {@see DynamicContent} inside cached content.
+ */
+final class CacheContent
+{
+    private string $id;
+    private CacheInterface $cache;
+    private CacheKeyNormalizer $cacheKeyNormalizer;
+
+    /**
+     * @var array<string, DynamicContent>
+     */
+    private array $dynamicContents = [];
+
+    /**
+     * @var string[]
+     */
+    private array $variations = [];
+
+    /**
+     * @param string $id The unique identifier of the cached content.
+     * @param CacheInterface $cache The cache instance.
+     * @param array $dynamicContents The dynamic content instances.
+     * @param array $variations List of string factors that would cause the variation of the content being cached.
+     */
+    public function __construct(string $id, CacheInterface $cache, array $dynamicContents = [], array $variations = [])
+    {
+        $this->id = $id;
+        $this->cache = $cache;
+        $this->cacheKeyNormalizer = new CacheKeyNormalizer();
+        $this->setDynamicContents($dynamicContents);
+        $this->setVariations($variations);
+    }
+
+    /**
+     * Caches, replaces placeholders with actual dynamic content, and returns the full actual content.
+     *
+     * @param string $content The content of the item to cache store.
+     * @param DateInterval|int|null $ttl The TTL of the cached content.
+     * @param Dependency|null $dependency The dependency of the cached content.
+     * @param float $beta The value for calculating the range that is used for "Probably early expiration".
+     *
+     * @see CacheInterface::getOrSet()
+     *
+     * @return string The rendered cached content.
+     */
+    public function cache(string $content, $ttl = 60, Dependency $dependency = null, float $beta = 1.0): string
```

Comment:
Does it make sense to call it multiple times? If not, maybe $content should be part of the constructor and the rest should be additional configuration methods?

Reply:
The `CacheContent ' object can be used inside a view file. And we need to have the placeholders separately, for use in the view file.